includes/abstracts/abstract-wc-log-handler.php

```diff
@@ -0,0 +1,135 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+if ( ! class_exists( 'WC_Logger' ) ) {
+	include_once( dirname( dirname( __FILE__ ) ) . '/class-wc-logger.php' );
+}
+
+/**
+ * Abstract WC Log Handler Class
+ *
+ * @version        2.8.0
+ * @package        WooCommerce/Abstracts
+ * @category       Abstract Class
+ * @author         WooThemes
+ */
+abstract class WC_Log_Handler {
+
+	/**
+	 * Minimum log level this handler will process.
+	 *
+	 * @var int 0-8 minimum level severity for handling log entry.
+	 * @access private
+	 */
+	protected $_threshold;
+
+	/**
+	 * Log levels by severity.
+	 *
+	 * @var array
+	 * @access private
+	 */
+	protected static $_log_levels = array(
+		WC_Logger::DEBUG     => 0,
+		WC_Logger::INFO      => 1,
+		WC_Logger::NOTICE    => 2,
+		WC_Logger::WARNING   => 3,
+		WC_Logger::ERROR     => 4,
+		WC_Logger::CRITICAL  => 6,
+		WC_Logger::ALERT     => 7,
+		WC_Logger::EMERGENCY => 8,
+	);
+
+	/**
+	 * Constructor for log handler.
+	 *
+	 * @param arr $args {
+	 *     @type string $threshold Optional. Default 'emergency'. Sets the log severity threshold.
+	 *         emergency|alert|critical|error|warning|notice|info|debug
+	 * }
+	 */
+	public function __construct( $args = array() ) {
+
+		$args = wp_parse_args( $args, array(
+			'threshold' => WC_Logger::EMERGENCY,
```

Comment:
I originally chose `emergency` because it was the least invasive default, which log handler implementers can then override with *their* default, and anyone using the logger should manually be able to set.

That said, my original idea is probably not the best. @gedex proposes a sensible default and @claudiosanches proposes log everything by default. For reference, monolog [seems to follow the log everything approach](https://github.com/Seldaek/monolog/blob/4a43d9b17c33bc889674f0fe6c8dc25a2b9b4f73/src/Monolog/Handler/AbstractHandler.php#L30), which seems logical to me. Filtering logs is a conscious decision, not an arbitrary default.